### PR TITLE
Add Target.cppParameterType, remove IN_GCC condition from cppmangle

### PR DESCRIPTION
**Pull request by ibuclaw**
~Won't succeed until #7619 and #7620 are merged.~

ETIMEDOUT on @WalterBright reviewing the other PR.

### CODE REVIEW

**Comment by wilzbach** [comment]
(rebased)

**Comment by ibuclaw** [comment]
Removed the dependency on #7620.  Moved the removal of `IN_GCC` to that PR.

File: src/dmd/target.d [hunk 1]
```diff
@@ -487,6 +489,25 @@ struct Target
         return null;
     }
 
+    /**
+     * Returns the type which will really be used for passing the given
+     * parameter to an extern(C++) function.
+     */
+    extern (C++) static Type cppParameterType(Parameter p)
```
Comment (JinShil) [inline src/dmd/target.d]: Should have a `Params` element n the DDoc header.

**Review by JinShil — CHANGES_REQUESTED** [review]

File: src/dmd/target.d (hunk 1 above)
Comment (JinShil) [reply src/dmd/target.d]: "really" sounds strange.  Did you mean "ultimately" or "eventually"?

Comment (ibuclaw) [reply src/dmd/target.d]: @JinShil - I'll ddoc the rest of target for consistency in another PR, as this module is my baby. ;-)

**Review by JinShil — APPROVED** [review]

File: src/dmd/target.d (hunk 1 above)
Comment (ibuclaw) [reply src/dmd/target.d]: Using "really" makes sense to me.  For example, a `lazy bool` will "really" be passed as a `bool delegate()`

**Comment by ibuclaw** [comment]
Rebased, again.

**Comment by ibuclaw** [comment]
Added `Returns:` to ddoc comment.

**Comment by ibuclaw** [comment]
Rebased again, re-adding auto-merge as has been re-added twice before.

**Pull request merged**